File: hospital-portal/src/app/services/global-setting.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

export interface GlobalSettingResponse {
  id: string;
  settingKey: string;
  settingValue: string;
  category: string;
  description: string;
  updatedBy: string;
  createdAt: string;
  updatedAt: string;
}

export interface GlobalSettingRequest {
  settingKey: string;
  settingValue: string;
  category?: string;
  description?: string;
}

@Injectable({ providedIn: 'root' })
export class GlobalSettingService {
  private readonly http = inject(HttpClient);

  list(category?: string): Observable<GlobalSettingResponse[]> {
    if (category) {
      return this.http.get<GlobalSettingResponse[]>('/super-admin/settings', {
        params: { category },
      });
    }
    return this.http.get<GlobalSettingResponse[]>('/super-admin/settings');
  }

  getByKey(settingKey: string): Observable<GlobalSettingResponse> {
    return this.http.get<GlobalSettingResponse>(
      `/super-admin/settings/${encodeURIComponent(settingKey)}`,
    );
  }

  upsert(req: GlobalSettingRequest): Observable<GlobalSettingResponse> {
    return this.http.put<GlobalSettingResponse>('/super-admin/settings', req);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`/super-admin/settings/${id}`);
  }
}

File: hospital-core/src/main/java/com/example/hms/service/GlobalSettingService.java
package com.example.hms.service;

import com.example.hms.payload.dto.globalsetting.GlobalSettingResponseDTO;

import java.util.List;
import java.util.UUID;

public interface GlobalSettingService {

    List<GlobalSettingResponseDTO> listAll();

    List<GlobalSettingResponseDTO> listByCategory(String category);

    GlobalSettingResponseDTO getByKey(String settingKey);

    GlobalSettingResponseDTO upsert(String settingKey, String settingValue, String category,
                                     String description, String updatedBy);

    void delete(UUID id, String deletedBy);
}
